integrationtest-service-framework/src/main/java/org/hypertrace/core/serviceframework/IntegrationTestServerUtil.java

```diff
@@ -25,22 +26,14 @@ public class IntegrationTestServerUtil {
   private static String[] services;
 
   public static void startServices(String[] services) {
-    executorService = Executors.newFixedThreadPool(services.length);
-    IntegrationTestServerUtil.services = services;
-    for (String service:services) {
-      executorService.submit(() -> {
-        try {
-          IntegrationTestServiceLauncher.main(new String[] {service});
-        } catch (Throwable t) {
-          System.out.println("Error starting the service with these message: "
-              + t.getMessage());
-        }
-      });
-      waitTillServerReady(service);
-    }
+    startServices(Optional.empty(), services);
   }
 
   public static void startServices(String testName, String[] services) {
+    startServices(Optional.of(testName), services);
+  }
+
+  public static void startServices(Optional<String> testName, String[] services) {
```

Comment:
nit - don't need to expose two ways of doing something, impl detail

```suggestion
  private static void startServices(Optional<String> testName, String[] services) {
```

Reply:
Done